src/nodal_properties.cc

```diff
@@ -42,3 +43,17 @@ void mpm::NodalProperties::update_property(
   properties_.at(property).block(node_id * nprops, mat_id, nprops, 1) =
       property_value + this->property(property, node_id, mat_id, nprops);
 }
+
+// Initialise all the nodal values for all properties in the property pool
+void mpm::NodalProperties::initialise_properties() {
+  // Iterate over all properties in the property map
+  for (std::map<std::string, Eigen::MatrixXd>::iterator prop_itr =
+           properties_.begin();
+       prop_itr != properties_.end(); ++prop_itr) {
+    // Create Matrix with zero values that has same size of the current property
+    // in the iteration
+    Eigen::MatrixXd zeroed_property =
```

Comment:
Is the size the entire size of particles * # of materials?

Reply:
The size is equal to rows * cols where:
- rows = the number of nodes * property dimension (1 if scalar, 2 or 3 if 2D or 3D vector, respectively)
- cols = number of materials